File: Java/cross_validate.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.util.StringTokenizer;

public class cross_validate {
    public static void main(String argv[]) {
	try {
	    for (int lay=0; lay<100; lay++) {
		for (int sc=0; sc<9; sc++) {
		    String currentLine;
		    BufferedReader br = new BufferedReader(new FileReader("../TestLayouts/layout"+lay+".txt"));
		    int i=0;
		    currentLine=br.readLine();
		    int n=Integer.parseInt(currentLine);
		    double layout[][]=new double[n][2];
		    while ((currentLine=br.readLine()) != null && i<n) {
			StringTokenizer tokenizer=new StringTokenizer(currentLine);
			int j=0;
			while (tokenizer.hasMoreElements() && j<2) {
			    layout[i][j]=Double.parseDouble(tokenizer.nextToken());
			    j++;
			}
			i++;
		    }
		    WindScenario ws = new WindScenario("../Scenarios/0"+sc+".xml");
		    KusiakLayoutEvaluator le = new KusiakLayoutEvaluator();
		    le.initialize(ws);
		    le.evaluate(layout);
		    //	    System.out.println("Energy output: "+le.getEnergyOutput());
		    System.out.println(le.getWakeFreeRatio());
		}
	    }
	} catch (Exception e) {
	    e.printStackTrace();
	}
    }
}

File: C++/cross_validate.cpp
//
//  main.cpp
//  TurbinesSimulatorTest
//
//  Created by Sylvain Cusat-Blanc on 04/12/12.
//  Copyright (c) 2012 Sylvain Cusat-Blanc. All rights reserved.
//

#include <iostream>
#include <time.h>
#include <unistd.h>
#include <ios>
#include <fstream>

#include "WindScenario.h"
#include "KusiakLayoutEvaluator.h"
#include "Matrix.hpp"


int main(int argc, const char * argv[]) {
  for (int i=0; i<100; i++) {
    for (int sc=0; sc<10; sc++) {
      stringstream layoutFileName;
      stringstream scFileName;
      char scnumber[64];
      sprintf(scnumber, "%02d", sc);
      layoutFileName << "../TestLayouts/layout" << i << ".txt";
      scFileName << "../Scenarios/" << scnumber << ".xml";

      ifstream layoutFile(layoutFileName.str().c_str());
      int n;
      layoutFile >> n;
      Matrix<double> tpos(n, 2);
      int current=0;
      if (layoutFile.is_open()) {
        double x, y;
        while (!layoutFile.eof()) {
          layoutFile >> x >> y;
          tpos.set(current, 0, x);
          tpos.set(current, 1, y);
          current++;
        }
      }
      layoutFile.close();
      WindScenario wsc(scFileName.str());
      KusiakLayoutEvaluator simu;
      simu.initialize(wsc);
      double wfr=simu.evaluate(&tpos);
      cout.precision(15);
      cout << wfr << endl;
    }
  }
}
